### ISSUE TITLE
Password is stored in clear text when using background mode

### ISSUE THREAD

**Issue by jverbru3** [larshp/abapGit#1887]
As mentioned on the settings page for background, password will be saved in clear text.
What about entering a HTTP RFC destination (type G) in the settings?

Then use cl_http_client=>create_by_destination instead of cl_http_client=>create_by_url in ZCL_ABAPGIT_HTTP->CREATE_BY_URL

**Comment by flaiker** [comment]
This could maybe already be possible, see #1853 #1841 #1848
I haven't looked at it in detail yet, it could be that it doesn't work for background mode.

**Comment by christianguenter2** [comment]
Yes, this is already possible with the userexit mentioned in the links. Just leave username and password empty.

**Comment by larshp** [comment]
@jverbru3 did it help, can this issue be closed?

**Comment by jverbru3** [comment]
Hi all, surely the comments did help, however I believe this is better a standard feature than something that has to be implemented via a user exit...  For sure for background mode. As the settings for background are already there, I think it's a small effort to include an option to fill an RFC destination instead of a user/pass?

**Comment by jverbru3** [comment]
the main reason to use the RFC is that the password is stored in the SAP secure area. If SAP could only give access to customers to use the secure area that would be really nice... maybe you can influence them ;-) for the URL part, that's created by the GIT code itself by setting request_uri in the header field.  Nevertheless, I got it working for background mode, just thought it would be nice to have this build into the standard.
